spec/test_diverse_word.rb

```diff
@@ -0,0 +1,86 @@
+require_relative '../solution/diverse_word'
+
+def checkResult(word, num_a, num_b, num_c)
+  return false if word.count('a') > num_a || word.count('b') > num_b || word.count('c') > num_c
+
+  (0...word.length - 1).each do |i|
+    return false if word[i] == word[i + 1] && word[i + 1] == word[i + 2]
+  end
+
+  true
+end
+
+RSpec.describe DiverseWord do
+  describe 'Example' do
+    it 'example1' do
+      test = DiverseWord.new(6, 1, 1).solution
+      expect(checkResult(test, 6, 1, 1)).to be true
+    end
+
+    it 'example2' do
+      test = DiverseWord.new(1, 3, 1).solution
+      expect(checkResult(test, 1, 3, 1)).to be true
+    end
+
+    it 'example3' do
+      test = DiverseWord.new(0, 1, 8).solution
+      expect(checkResult(test, 0, 1, 8)).to be true
+    end
+  end
+  describe 'Correctness test' do
+    it 'super_small' do
```

Comment:
Also leave a line break here.
```suggestion
    end
  end

  describe 'Correctness test' do
    it 'super_small' do
```